### ISSUE TITLE
Error when integrate sqrt

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#433]
Imported from SourceForge on 2024-07-02 19:49:46
Created by **[jddviramontes](https://sourceforge.net/u/jddviramontes/)** on 2014-07-03 19:20:25
Original: https://sourceforge.net/p/maxima/bugs/2776

---

When we integrate a square root like this:

~~~~

integrate ((1+exp(t))/sqrt(t+exp(t)), t, 0, 1);

~~~~

it do not provide any answer. 

But with the next example of the same degree of difficulty it provide a correct answer:

~~~~

integrate ((1+2*t)/sqrt(t+t^2), t, 0, 1);

~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-02 19:49:47
Created by **[tomio-arisaka](https://sourceforge.net/u/tomio-arisaka/)** on 2014-07-13 11:05:25
Original: https://sourceforge.net/p/maxima/bugs/2776/#8031

---

It is interesting that such definite integral is not computed,
But the indefinite integral seems to be computed.
The ldefint function can also compute such definite integral.

    Maxima 5.33.0 http://maxima.sourceforge.net
    using Lisp SBCL 1.1.17
    Distributed under the GNU Public License. See the file COPYING.
    Dedicated to the memory of William Schelter.
    The function bug_report() provides bug reporting information.
    (%i1) integrate ((1+exp(t))/sqrt(t+exp(t)), t, 0, 1);
                                   1
                                  /       t
                                  [     %e  + 1
    (%o1)                         I  ------------- dt
                                  ]         t
                                  /  sqrt(%e  + t)
                                   0


    (%i2) u(t) := %e^t+t;
                                              t
    (%o2)                           u(t) := %e  + t
    (%i3) 'integrate ('diff('u(t), t)/sqrt('u(t)), t);
                                      d
                                    / -- (u(t))
                                    [ dt
    (%o3)                           I ---------- dt
                                    ] sqrt(u(t))
                                    /

The indefinite integral:

    (%i4) integrate ((1+exp(t))/sqrt(t+exp(t)), t);
                                             t
    (%o4)                           2 sqrt(%e  + t)
    (%i5) ev(%, t:1) - ev(%, t:0);
    (%o5)                         2 sqrt(%e + 1) - 2

The ldefint:

    (%i7) ldefint ((1+exp(t))/sqrt(t+exp(t)), t, 0, 1);
    (%o7)                         2 sqrt(%e + 1) - 2

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-02 19:49:49
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2014-07-26 06:31:26
Original: https://sourceforge.net/p/maxima/bugs/2776/#686c

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-02 19:49:51
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2014-07-26 06:31:27
Original: https://sourceforge.net/p/maxima/bugs/2776/#cb4f

---

fixed by [b0776c24bf74d0cf909055600cd1f6594c303c2c]

    src/defint.lisp:
    o poles-in-interval: test sign of denom.  This handles cases where we
      can't find the exact roots, but we know that they occur outside the
      interval of integration.
      example: integrate((1+exp(t))/sqrt(t+exp(t)), t, 0, 1);
